metadata/src/main/java/org/apache/kafka/controller/metrics/ControllerMetadataMetrics.java

```diff
@@ -157,6 +200,30 @@ public void addToActiveBrokerCount(int brokerCountDelta) {
     public int activeBrokerCount() {
         return this.activeBrokerCount.get();
     }
+
+    public void setControlledShutdownBrokerCount(int brokerCount) {
+        this.controllerShutdownBrokerCount.set(brokerCount);
+    }
+
+    public void addToControlledShutdownBrokerCount(int brokerCountDelta) {
+        this.controllerShutdownBrokerCount.addAndGet(brokerCountDelta);
+    }
+
+    public int controlledShutdownBrokerCount() {
+        return this.controllerShutdownBrokerCount.get();
+    }
+
+    public void setBrokerRegistrationState(int brokerId, int brokerState) {
```

Comment:
we should be able to just pass the `BrokerRegistration` object to `setBrokerRegistrationState`, right? Or more likely pass the enum to here, and have a function to create the enum based on the `BrokerRegistration` object.